packages/file-search/src/node/file-search-service-impl.spec.ts

```diff
@@ -118,4 +120,35 @@ describe('search-service', function () {
             expect(matches.length).to.eq(0);
         });
     });
+
+    describe('irrelevant absolute results', () => {
+        const rootUri = FileUri.create(path.resolve(__dirname, '../../../../..'));
+
+        it('not fuzzy', async () => {
```

Comment:
here is a repro for the bogus `workspace` search results. 

```
        const rootUri = FileUri.create(path.resolve(__dirname, '../../../..'));

        it.only('not fuzzy', async () => {
            const searchPattern = rootUri.path.dir.base;
            const matches = await service.find(searchPattern, { rootUris: [rootUri.toString()], fuzzyMatch: false, useGitIgnore: true, limit: 200 });
            for (const match of matches) {
                const relativUri = rootUri.relative(new URI(match));
                if (relativUri) {
                    const relativMatch = relativUri.toString();
                    assert.notEqual(relativMatch.indexOf(searchPattern), -1, relativMatch);
                }
            }
        });
```

Reply:
@elaihau, ☝️ the `workspace` search pattern is in the path of the bogus search results.

Reply:
`it.only` nice feature, did not know it :)

Reply:
just don't forget to remove it ;-)

Reply:
It's because of glob search: https://github.com/theia-ide/theia/blob/8235400c78af8eb46ef5523ae1a635ec9761a0c3/packages/file-search/src/node/file-search-service-impl.ts#L93-L94

@elaihau @vince-fugnitto @AlexTugarev do you have an idea why we do it in addition to fuzzy search?

Reply:
we do it because vscode supports it and vscode extensions pass globs to the file-search backend. if our goal is supporting vscode extensions we should have it too. 

would searching a glob of "*workspace*" exclude results such as "search-in-workspace-server.ts" ?

Reply:
or ... maybe globs are written in different ways in the gitpod os ?

Reply:
We should provide cwd to rigrep to limit the search scope. I will rework a PR.

Reply:
We also track results separately for global and fuzzy matches, so it can return over limit 🙈

Reply:
the different is how you spawn Theia, rigrep seems to use process.cwd of Theia and if you spawn Theia from different folders, one gets different results. We should make sure that it is not important and configure rigrep with proper cwd, i.e. root URI. then it will look only inside a root and won't report irrelevant files.

Reply:
nice catch!

Reply:
Nice catch, thank you !

> We also track results separately for global and fuzzy matches, so it can return over limit

No we will not. in the code i created a Set to eliminate duplicates, sort, and slice to remove the ones that are above the limit :)

Reply:
> we do it because vscode supports it and vscode extensions pass globs to the file-search backend. if our goal is supporting vscode extensions we should have it too.

@elaihau @vince-fugnitto Could you explain me how VS Code use glob? I've analysed their source code and don't see that they do it. Which use cases are possible with it?

Reply:
ah, I see it is for vscode extension, i don't think API was modeled properly, there should be a new explicit option for included globs and it should not affect quick file

Reply:
@akosyakov  this is one example: https://github.com/Microsoft/vscode/blob/master/extensions/npm/src/tasks.ts#L101

npm uses the service to find all `packages.json`, excluding those from `node_modules`.

Reply:
there are a few examples tested here as well:

https://github.com/Microsoft/vscode/blob/92629f3f5f4c662a26cc193b5208c3ff3a335243/extensions/vscode-api-tests/src/singlefolder-tests/workspace.test.ts#L490-L522

Reply:
> there should be a new explicit option for included globs and it should not affect quick file

i was thinking about the same thing. but after seeing this interface i chose to support the string match & glob match from the same function. https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/services/search/common/search.ts#L79
```ts
export interface IFileQueryProps<U extends UriComponents> extends ICommonQueryProps<U> {
	type: QueryType.File;
	filePattern?: string;

	/**
	 * If true no results will be returned. Instead `limitHit` will indicate if at least one result exists or not.
	 * Currently does not work with queries including a 'siblings clause'.
	 */
	exists?: boolean;
	sortByScore?: boolean;
	cacheKey?: string;
}
```
maybe you are right, with the new option defined we would have finer grained control over what the service does.

Reply:
@elaihau yes, `searchPattern` and `includePatterns` are not the same, i've separated them